### ISSUE TITLE
Turning switch on from Home Assistant

### ISSUE THREAD

**Issue by PhillyGilly** [MindrustUK/Heatmiser-for-home-assistant#113]
I'm probably missing something really obvious but.....
I want to turn on and off a heatmiser neo-HW using an automation in HA.
I set up a profile in the heatmiser app which has a name but no ON or OFF times set, then I a simple test automation in HA:
```alias: Towel Rail timed
description: ""
trigger:
  - platform: time
    at: "10:00:00"
    id: towel1_1
condition: []
action:
  - choose:
      - conditions:
          - condition: trigger
            id: towel1_1
        sequence:
          - service: switch.turn_on
            data: {}
            target:
              entity_id:
                - switch.towel_rail_1
          - delay:
              hours: 1
              minutes: 0
              seconds: 0
              milliseconds: 0
          - service: switch.turn_off
            data: {}
            target:
              entity_id: switch.towel_rail_1
```
What happens is that the switch status in HA updates but the status in the Heatmiser App remains the same then after a short while then goes off again.
Any help appreciated.

**Comment by ribbal** [comment]
Hmmmm…….this works for me (though I’m not using any profiles).  If you manually flick the switch on the HA dashboard, does the status update in the Heatmiser app?

**Comment by PhillyGilly** [comment]
If I operate the switch on my HA dashboard, it changes the state shown in the app for about 60secs before it switched off again.

**Comment by ribbal** [comment]
Hmmmm……this works as expected for me.  Are you running the very latest version of the Heatmiser integration?

**Comment by PhillyGilly** [comment]
Yes, as far as I know. Was this a recent change? I'll have to check/confirm later.

**Comment by ribbal** [comment]
No, I don't believe there have been any recent changes.

**Comment by PhillyGilly** [comment]
@ribbal I looked in HACS but couldn't see what version I am running. I ran the package update and the number here doesn't change. Does this mean I am on the latest?
![image](https://user-images.githubusercontent.com/56273663/201861720-1540c11e-68e3-493c-a82d-342bbff90974.png)

**Comment by ribbal** [comment]
Yes, you are on the latest version.

**Comment by petitpiton** [comment]
I have a similar problem. I have an automation to control the Neo for Hot Water (Gas Boiler).  The automation triggers when the temperature of the tank drops below a threshold level and switches the Hot Water on.  It was working as expected until recently. Now, the automation switches the hot water on, but 30 seconds later the switch reverts to off (no HA service tagged as having done this), so the automation triggers again and 30 seconds later the hot water switches itself off again, and so on.
I haven't made any changes to the automation and have only applied core updates to HA (2023.2.5 currently).
Screenshot to show the behaviour described above.
<img width="918" alt="Screenshot 2023-02-21 at 10 28 35" src="https://user-images.githubusercontent.com/63656369/220319706-1123edc5-24d6-458d-b382-b11c27801200.png">
Any ideas?

**Comment by PhillyGilly** [comment]
Just come back to this as I wanted to turn on my towel rails when there is excess solar PV power. It feels to me as though the action is just taking place in HA but never getting sent to the Heatmiser so the HA display shows the status base don the command but reverts to the actual Heatmiser status on the next scan.

**Comment by coathook** [comment]
Hi, I have a Neostat Air which is used for controlling the landing temperature, and the time switch function for hot water
I see a similar issue, that Home Assistant turns the switch on, and then off again a few seconds later, and then on again. 
from the history:

<img width="656" alt="image" src="https://github.com/MindrustUK/Heatmiser-for-home-assistant/assets/127053710/4230653c-8b7c-48d1-b203-7f5cc9da13ff">

is this to be expected?  It does turn on and work but the state shows "On-Off-On"

**Comment by DKen71** [comment]
I have a similar situation to @coathook second issue. I have two NeoStat units that act as switches to turn on electric towel radiators. I have set up automations to toggle these switches at a set time. The automation fires and turns the switch on. However, Heatmiser interprets this as a boost and sets the timer for 30 minutes, after which the switch toggles back off. I have looked at the **Heatmiser Neo Climate: Temperature Hold On** call service, but that only recognises climate entities within Heatmiser and not any switches within heatmiser. Is there a way to toggle this switch between standby and auto or to set the hold time, in the way that we can with climate entities.

**Comment by petitpiton** [comment]
@DKen71
If you’re comfortable editing python code, you can modify the switch.py file at around line 54:

        self._holdfor = 30

I changed this to 60 and now the boosts are for an hour.
——————
Regards
Dr David Johnson

**Comment by PhillyGilly** [comment]
> Hi, I have a Neostat Air which is used for controlling the landing temperature, and the time switch function is used for hot water I see a similar issue, that Home Assistant turns the switch on, and then off again a few seconds later, and then on again. from the history:
> 
> <img alt="image" width="656" src="https://user-images.githubusercontent.com/127053710/258589963-4230653c-8b7c-48d1-b203-7f5cc9da13ff.png">
> is this to be expected?
> 
> Secondly the "On" command actually triggers the Neostat to switch on for 30mins (like a boost) so for 1 hr of hot water I trigger two 30min cycles. Is there any way to change the 30mins timeout from HA to e.g. 60mins? - that would be very helpful in a future update of the integration

Could you explain “0n” command, pls?

**Comment by bonkersbrains** [comment]
I have the same problem, I switch the hot water on manually in HA and a few seconds later it switches off on its own.

**Comment by bonkersbrains** [comment]
I agree that the heatmiser cloud is messing things up. In my case I have a profile for central heating, and HA does not get involved in that. 
And I have no profile for hot water, and I control it only through HA. As I said above, with the changes I made it seems to be ok now (even though it turns on, then off, then on again). I have updated the hold time to 60 mins in switch.py line 54 and i think it should be ok now.
